conans/test/functional/graph_lock/graph_lock_test.py

```diff
@@ -368,11 +368,11 @@ def export_pkg_test(self):
 class GraphLockBuildRequireVersionRangeTest(GraphLockVersionRangeTest):
     consumer = GenConanfile().with_name("PkgB").with_version("0.1")\
                              .with_build_require_plain("PkgA/[>=0.1]@user/channel")
-    pkg_b_revision = "b6f49e5ba6dd3d64af09a2f288e71330"
+    pkg_b_revision = "b7338f650cc61f8e0ad0285cd1b77b92"
```

Comment:
this might be too fragile that package ids change, but if there is no better way to manage, it's okay

Reply:
The issue here is that the generated ``conanfile.py`` with ``GenConanfile()`` has changed, moving from ``build_requires`` attribute to ``build_requirements()`` method. This is very expected and desired, we need to be aware that revisions change when something changed, it is a good thing that this test failed.

I would have preferred that ``GenConanfile()`` didn't change, and users could explicitly select if they want to use the attribute or the method, and this test didn't change.

Reply:
* If there is none, let's add a test to check that if the `conanfile` changes the revision is different... it should be already tested.
* This test should use plain conanfiles, in this test it is an undesired side-effect (and a PITA to fix).
* About `GenConanfile` itself, good news is that changing this doesn't break other tests 😄